pkg/midstream/identity.go

```diff
@@ -0,0 +1,175 @@
+package midstream
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"text/template"
+
+	"github.com/dexidp/dex/server"
+	dexstorage "github.com/dexidp/dex/storage"
+	"github.com/pkg/errors"
+	kotsv1beta1 "github.com/replicatedhq/kots/kotskinds/apis/kots/v1beta1"
+	identitydeploy "github.com/replicatedhq/kots/pkg/identity/deploy"
+	dextypes "github.com/replicatedhq/kots/pkg/identity/types/dex"
+	"github.com/replicatedhq/kots/pkg/ingress"
+	"github.com/replicatedhq/kots/pkg/k8sutil"
+	"github.com/segmentio/ksuid"
+	yaml "gopkg.in/yaml.v2"
+	kustomizetypes "sigs.k8s.io/kustomize/api/types"
+)
+
+func (m *Midstream) writeIdentityService(ctx context.Context, options WriteOptions) (string, error) {
+	if m.IdentityConfig == nil {
+		return "", nil
+	}
+
+	dexConfig, err := getDexConfig(ctx, m.IdentityConfig.Spec)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get dex config")
+	}
+
+	resources, err := identitydeploy.Render(ctx, options.AppSlug, dexConfig, m.IdentityConfig.Spec.IngressConfig, nil)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to render identity service")
+	}
+
+	kustomization := kustomizetypes.Kustomization{
+		TypeMeta: kustomizetypes.TypeMeta{
+			APIVersion: "kustomize.config.k8s.io/v1beta1",
+			Kind:       "Kustomization",
+		},
+		CommonLabels: map[string]string{
```

Comment:
Using common labels will break subsequent deployments when adding additional labels later, because kustomize also tries to inject those labels into `matchLabels` which is immutable. We used label transformers to resolve this for disaster recovery. https://github.com/replicatedhq/kots/blob/359ea2ab3007b19dc69ac02accda824e6d79145f/pkg/disasterrecovery/disasterrecovery.go#L23

Reply:
thanks